File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/TeamStateEventId.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\StringIdInterface;

class TeamStateEventId implements StringIdInterface
{
    /**
     * @Type("string")
     */
    private $teamName;

    /**
     * @return string
     */
    public function getTeamName()
    {
        return $this->teamName;
    }

    /**
     * @param string $teamName
     */
    public function setTeamName($teamName)
    {
        $this->teamName = $teamName;
    }

    /**
     * @Type("Dddml\Serializer\Type\Long")
     */
    private $version;

    /**
     * @return Long
     */
    public function getVersion()
    {
        return $this->version;
    }

    /**
     * @param Long $version
     */
    public function setVersion($version)
    {
        $this->version = $version;
    }



    /**
     * @var TeamStateEventIdFlattenedDto
     */
    private $idFlattenedDto;

    /**
     * @return string
     */
    public function toString()
    {
        if (!$this->idFlattenedDto) {
            $this->idFlattenedDto = new TeamStateEventIdFlattenedDto($this);
        }

        return $this->idFlattenedDto->toString();
    }

    /**
     * @param string $idStr
     *
     * @return TeamStateEventId
     */
    public static function createFromString($idStr)
    {
        return (new TeamStateEventIdFlattenedDto())
            ->fromString($idStr)
            ->toTeamStateEventId();
    }


}

File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/TeamStateEventId.java
package org.dddml.wms.domain;

import java.io.Serializable;

public class TeamStateEventId implements Serializable
{
    private String teamName;

    public String getTeamName()
    {
        return this.teamName;
    }

    public void setTeamName(String teamName)
    {
        this.teamName = teamName;
    }

    private Long version;

    public Long getVersion()
    {
        return this.version;
    }

    public void setVersion(Long version)
    {
        this.version = version;
    }

    public TeamStateEventId()
    {
    }

    public TeamStateEventId(String teamName, Long version)
    {
        this.teamName = teamName;
        this.version = version;
    }

    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        TeamStateEventId other = (TeamStateEventId)obj;
        return true 
            && (teamName == other.teamName || (teamName != null && teamName.equals(other.teamName)))
            && (version == other.version || (version != null && version.equals(other.version)))
            ;
    }

    @Override
    public int hashCode()
    {
        int hash = 0;
        if (this.teamName != null) {
            hash += 13 * this.teamName.hashCode();
        }
        if (this.version != null) {
            hash += 13 * this.version.hashCode();
        }
        return hash;
    }

}

